great_expectations/core/config_peer.py

```diff
@@ -44,26 +41,26 @@ def config(self) -> BaseYamlConfig:
 
     def get_config(
         self,
-        mode: ConfigOutputModeType = ConfigOutputModes.TYPED,
+        mode: ConfigOutputModes = ConfigOutputModes.TYPED,
         **kwargs,
     ) -> Union[BaseYamlConfig, dict, str]:
         if isinstance(mode, str):
             mode = ConfigOutputModes(mode.lower())
 
         config: BaseYamlConfig = self.config
 
-        if mode == ConfigOutputModes.TYPED:
+        if mode is ConfigOutputModes.TYPED:
```

Comment:
Do we want this to evaluate as True for a str equivalent?
I know that it will, I'm just trying to understand the intent here.

Reply:
ah wait yes maybe `==` is proper? I don't think this is external facing so I'd honestly prefer a plain old enum to be honest

Reply:
I believe `is` will actually work between 2 equivalent strings, but that's just a side effect of some `CPython`optimizations, but it's not something we should rely on. It's the same when comparing some `int`s, but stops working after like `256`.

I think the `str` enum mixn is fine. But I would keep this an equality rather than an identity check.